### PR TITLE
Make SSL optional when cert is set for ingestor and kibana

### PR DESCRIPTION
**Pull request by janaurka**
This introduces `properties.haproxy.kibana.enable_ssl` and
`properties.haproxy.ingestor.enable_ssl` (both defaults to false)
which take care of enable SSL on the haproxy more flexible then
before.

Previously if a SSL cert was given, port `5514` (ingestor) and
kibana, port 80 (HTTP) or 443 (HTTPS) was automatically enabled.

With this commit, one can be enabled when the other is disabled.

Example:
Ingestor (syslog) traffic should be plainext, but the
Kibana traffic should be encrypted.

### CODE REVIEW

File: jobs/haproxy/templates/haproxy.config.erb [hunk 1]
```diff
@@ -86,14 +86,17 @@ backend cluster_monitor
 frontend elastic-in
     mode http
 
-    <% if properties.haproxy.ssl_pem %>
+    <% if properties.haproxy.ssl_pem and properties.haproxy.elastic.enable_ssl == true %>
```
Comment (Infra-Red) [inline jobs/haproxy/templates/haproxy.config.erb:110]: This property is missing in haproxy job spec.

**Review by Infra-Red — CHANGES_REQUESTED** [review]

**Comment by janaurka** [comment]
@Infra-Red: You are right, should be fixed with 7ab1ecb.

File: jobs/haproxy/spec [hunk 2]
```diff
@@ -69,6 +75,9 @@ properties:
   haproxy.elastic.backend_servers:
     description: "Lish of Elasticsearch nodes"
     default: []
+  haproxy.elastic.enable_ssl:
+    description: "Enable or Disable SSL for for syslog/ingestor traffic on haproxy"
```
Comment (Infra-Red) [inline jobs/haproxy/spec]: @janaurka Could you please also update property description?

**Review by Infra-Red — CHANGES_REQUESTED** [review]

**Comment by janaurka** [comment]
@Infra-Red: sorry, this somehow went unnoticed. Should be fixed now.

**Review by Infra-Red — APPROVED** [review]

**Pull request merged**